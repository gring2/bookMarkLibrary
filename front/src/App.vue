<template>
  <div id="app">
    <Header ></Header>
    <router-view/>
  </div>
</template>


<script lang="ts">
  import { Component, Vue } from 'vue-property-decorator'
  import Header from '@/views/Header.vue' // @ is an alias to /src
  import User from '@/vo/User'
  import { mapState } from 'vuex'
@Component({
    components: {
      Header,
    },
  })

export default class App extends Vue {
}
</script>

<style lang="scss">
  @import '@/styles/init.scss'

  /*#app {*/
/*  font-family: 'Avenir', Helvetica, Arial, sans-serif;*/
/*  -webkit-font-smoothing: antialiased;*/
/*  -moz-osx-font-smoothing: grayscale;*/
/*  text-align: center;*/
/*  color: #2c3e50;*/
/*}*/
/*#nav {*/
/*  padding: 30px;*/
/*  a {*/
/*    font-weight: bold;*/
/*    color: #2c3e50;*/
/*    &.router-link-exact-active {*/
/*      color: #42b983;*/
/*    }*/
/*  }*/
/*}*/
</style>
