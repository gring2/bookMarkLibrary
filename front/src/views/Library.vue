<template>
  <section class="content">
    <div>
      <TagsPanel></TagsPanel>
      <BookmarksPanel></BookmarksPanel>
    </div>

  </section>
</template>

<script lang="ts">
  import { Component, Prop, Vue } from 'vue-property-decorator'
  import TagsPanel from '@/views/TagsPanel.vue'
  import BookmarksPanel from '@/views/BookmarksPanel.vue'
  import {bookmarkMod} from '@/stores/modules/bookmark'

  @Component({
    components: {
      TagsPanel,
      BookmarksPanel
    },
  })
  export default class Library extends Vue {
    created(){
      bookmarkMod.GET_BOOKMARKS()
    }

  }
</script>

<!--<style scoped lang="scss">-->
<!--  @import "@/styles/header.scss"-->
<!--</style>-->

<style scoped lang="scss">
  @import "@/styles/library/urls.scss";
</style>

