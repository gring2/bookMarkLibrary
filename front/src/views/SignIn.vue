<template>
  <section class="content">
    <SignInForm/>
  </section>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import SignInForm from '@/components/SignInForm.vue'
@Component({
  components: {
    SignInForm
  },
})
export default class SignIn extends Vue {}
</script>

<style lang="scss" scoped>
  section {
    display: flex;
    justify-content: center;
    align-items: center
  }
</style>

