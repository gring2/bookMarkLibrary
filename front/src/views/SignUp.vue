<template>
  <section>
    <SignUpForm/>
  </section>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import SignUpForm from '@/components/SignUpForm.vue'
@Component({
  components: {
    SignUpForm
  },
})
export default class SignUp extends Vue {}
</script>
