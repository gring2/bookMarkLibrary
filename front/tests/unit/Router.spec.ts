import { mount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Library from '@/views/Library.vue'
import SignUp from '@/views/SignUp.vue'
import SignIn from '@/views/SignIn.vue'
import App from '@/App.vue'

describe('Router rendering views', () => {
  let localVue: any;
  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VueRouter)

  })

  it('renders a SignUp Views via routing', () => {
    const router = new VueRouter({
      mode: 'history',
          base: process.env.BASE_URL,
          routes: [
            {
              path: '/signup',
              name: 'signup',
              component: SignUp,
            },
      ],
    })
    SignUp
    const wrapper = mount(App, { localVue, router })

    router.push('/signup')
    wrapper.vm.$nextTick(() => {
      expect(wrapper.find(SignUp).exists()).toBeTruthy()
      expect(wrapper.element).toMatchSnapshot()
  
    })
  })

  it('renders a Library view via routing', () => {

    const router = new VueRouter({
      mode: 'history',
          base: process.env.BASE_URL,
          routes: [
            {
              path: '/bookmarks',
              name: 'bookmarks',
              component: Library,
            },
      ],
    })

    const wrapper = mount(App, { localVue, router })

    router.push('/bookmarks')
    expect(wrapper.find(Library).exists()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders a SignIn view via routing', () => {

    const router = new VueRouter({
      mode: 'history',
          base: process.env.BASE_URL,
          routes: [
            {
              path: '/signin',
              name: 'signin',
              component: SignIn,
            },
      ],
    })

    const wrapper = mount(App, { localVue, router })

    router.push('/signin')
    expect(wrapper.find(SignIn).exists()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

})