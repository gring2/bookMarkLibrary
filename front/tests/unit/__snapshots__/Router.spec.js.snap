// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router rendering views renders a Library view via routing 1`] = `
<div
  id="app"
>
  <nav>
    <div>
      <div
        class="home-logo"
      >
        <a
          class="link-style-none router-link-active"
          href="/"
        >
          <h1>
            BookMarkLibrary
          </h1>
        </a>
      </div>
       
      <div>
        <ul
          class="list-style-none flex flex-items-center"
        >
          <li>
            <a
              class="manu router-link-exact-active router-link-active"
              href="/bookmarks"
            >
              library
            </a>
          </li>
        </ul>
      </div>
    </div>
     
    <!---->
  </nav>
   
  <section
    class="content"
  >
    <div>
      <aside
        class="side"
      />
       
      <main>
        <div>
          <input
            placeholder="url"
            style="width: 100%; font-size: 2rem; height: 2rem;"
            type="text"
          />
           
          <input
            placeholder="#tag"
            style="width: 100%; font-size: 2rem; height: 2rem;"
            type="text"
          />
           
          <button
            class="submit"
            style="font-weight: normal; font-size: 1.4rem; width: 8rem;"
          >
            Susbmit
          </button>
        </div>
         
        <div
          class="container"
        >
          <ul />
        </div>
      </main>
    </div>
  </section>
</div>
`;

exports[`Router rendering views renders a SignIn view via routing 1`] = `
<div
  id="app"
>
  <nav>
    <div>
      <div
        class="home-logo"
      >
        <a
          class="link-style-none router-link-active"
          href="/"
        >
          <h1>
            BookMarkLibrary
          </h1>
        </a>
      </div>
       
      <div>
        <ul
          class="list-style-none flex flex-items-center"
        >
          <li>
            <a
              class="manu router-link-active"
              href="/"
            >
              library
            </a>
          </li>
        </ul>
      </div>
    </div>
     
    <!---->
  </nav>
   
  <section
    class="content"
  >
    <div>
      <div>
        <label>
          Email: 
        </label>
         
        <input
          type="text"
        />
      </div>
       
      <div>
        <label>
          password: 
        </label>
         
        <input
          type="password"
        />
      </div>
       
      <div>
        <button
          class=""
        >
          Susbmit
        </button>
      </div>
    </div>
  </section>
</div>
`;

exports[`Router rendering views renders a SignUp Views via routing 1`] = `
<div
  id="app"
>
  <nav>
    <div>
      <div
        class="home-logo"
      >
        <a
          class="link-style-none router-link-active"
          href="/"
        >
          <h1>
            BookMarkLibrary
          </h1>
        </a>
      </div>
       
      <div>
        <ul
          class="list-style-none flex flex-items-center"
        >
          <li>
            <a
              class="manu router-link-active"
              href="/"
            >
              library
            </a>
          </li>
        </ul>
      </div>
    </div>
     
    <!---->
  </nav>
   
  <section
    class="content"
  >
    <div>
      <div>
        <label>
          Email: 
        </label>
         
        <input
          type="text"
        />
      </div>
       
      <div>
        <label>
          password: 
        </label>
         
        <input
          type="password"
        />
      </div>
       
      <div>
        <label>
          password confirm: 
        </label>
         
        <input
          type="password"
        />
      </div>
       
      <div>
        <button
          class=""
        >
          Susbmit
        </button>
      </div>
    </div>
  </section>
</div>
`;
